### DISCUSSION TITLE
Error building android to run tests

### DISCUSSION THREAD

**Discussion by thanksyouall** [kirillzyusko/react-native-keyboard-controller#150] [General]
I am creating an android build to run detox tests. The command is `./gradlew assembleDebug assembleAndroidTest` or `/gradlew assembleDebug assembleAndroidTest -DtestBuildType=debug`.

And I get an error:
```
Execution failed for task ':react-native-keyboard-controller:processDebugAndroidTestManifest'.
> Manifest merger failed : uses-sdk:minSdkVersion 16 cannot be smaller than version 21 declared in library [com.facebook.react:react-android:0.71.8] /Users/ivan/.gradle/caches/transforms-3/2d0ccc383ad42b083a7f9f1673ca7c31/transformed/jetified-react-android-0.71.8-debug/AndroidManifest.xml as the library might be using APIs not available in 16
        Suggestion: use a compatible library with a minSdk of at most 16,
                or increase this project's minSdk version to at least 21,
                or use tools:overrideLibrary="com.facebook.react" to force usage (may lead to runtime failures)
```

This error didn't happen until I installed the `react-native-keyboard-controller` library. What could be the problem?

**android/build.gradle:**
```
buildscript {
    ext {
        buildToolsVersion = "33.0.0"
        minSdkVersion = 21
        compileSdkVersion = 33
        targetSdkVersion = 33

        // Мы используем NDK 23, который имеет поддержку M1 и является версией NDK от AGP.
        ndkVersion = "23.1.7779620"

        kotlinVersion = "1.6.21"
    }
    ...
}
```

**android/app/build.gradle:**
```
...
defaultConfig {
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 1
        versionName "1.0"
        ...
}
...
```

**Comment by thanksyouall** [comment]
@kirillzyusko or would it be better to create an issue?

**Reply by kirillzyusko** [reply]
@thanksyouall please, create issue :)

This library requires minSdk=21, because as far as I know current react-native version supports only 21+.

Also this library doesn't work in sdk < 21 (I mean it will not have any effect).

Let me know if you think I need to declare minSdk as 16 :)

**Reply by kirillzyusko** [reply]
🥺 
Just realized this library declares support for 16 
I think I need to change it to 21. Can you please patch build.gralde of this library and tell me if the issue is gone?

**Reply by kirillzyusko** [reply]
@thanksyouall could you please try to use this https://github.com/kirillzyusko/react-native-keyboard-controller/pull/151 in your project? I believe it should fix your problem 🙂 

And yes, thank you for raising the issue 🙌

**Reply by thanksyouall** [reply]
@kirillzyusko thank you! Yes, it works. You can merge that PR :)

**Reply by thanksyouall** [reply]
@kirillzyusko Could you please up the version?

**Reply by kirillzyusko** [reply]
@thanksyouall yes, I'll do it. I just want to do more research on react-native-navigation bug and maybe fix it and then release. If not - then will release without react-native-navigaton fix. I think I'll publish new releases in next few days :)

As of now - please, use version from main branch (or use patch-package - anything you prefer) 🙂

**Reply by thanksyouall** [reply]
@kirillzyusko thanks :-)

**Reply by kirillzyusko** [reply]
@thanksyouall I've published `1.5.3` 🙂
